jobs/release-microk8s/release-microk8s.groovy

```diff
@@ -0,0 +1,131 @@
+                    } else {
+                        all_tracks = [params.ARCH_TRACK.tokenize('/').last()]
+                    }
+                }
+            }
+        }
+        stage("Snapcraft login") {
+            steps {
+                sh "snapcraft login --with ${SNAPCRAFTCREDS}"
+            }
+        }
+        stage("Setup tox environment") {
+            steps {
+                sh """
+                tox -e py38 -- python -c 'print("Tox Environment Ready")'
+                """
+            }
+        }
+        stage("Release Tracks") {
+            steps {
+                script {
+                    all_tracks.each { track -> 
+                        stage("Track ${track}") {
+                            script {
+                                def juju_controller="release-microk8s-${track}-${env.ARCH}"
+                                def juju_model="release-microk8s-${track}-model"
+                                def juju_full_model="${juju_controller}:${juju_model}"
+                                def instance_type = ""
+                                def constraints = ""
+                                if (env.ARCH == "arm64") {
+                                    instance_type = "a1.2xlarge"
+                                    constraints = "instance-type=${instance_type} arch=$ARCH root-disk=80G"
+                                } else if (env.ARCH == "amd64") {
+                                    instance_type = "m5.large"
+                                    constraints = "instance-type=${instance_type} arch=$ARCH root-disk=80G mem=16G cores=8"
+                                } else {
+                                    error("Aborting build due to unknown arch=${env.ARCH}")
+                                }
+                                sh """
+                                if ! timeout 4m juju destroy-controller -y --destroy-all-models --destroy-storage "${juju_controller}"; then
+                                   timeout 4m juju kill-controller -y "${juju_controller}" || true
+                                fi
+                                """
+                                sh """
+                                juju bootstrap "${env.JUJU_CLOUD}" "${juju_controller}" \
+                                    -d "${juju_model}" \
+                                    --model-default test-mode=true \
+                                    --model-default resource-tags=owner=k8sci \
+                                    --bootstrap-constraints "instance-type=${instance_type}"
+
+                                juju deploy -m "${juju_full_model}" --constraints "${constraints}" ubuntu
+
+                                juju-wait -e "${juju_full_model}" -w
+
+                                juju ssh -m "${juju_full_model}" --pty=true ubuntu/0 -- 'sudo snap install lxd'
+                                juju ssh -m "${juju_full_model}" --pty=true ubuntu/0 -- 'sudo lxd.migrate -yes' || true
```

Comment:
previous ogs scripts were running with `set +e` which would  not error on bash errors, whereas jenkins runs with `set -e` be default.  this line seems to fail on jenkins every time but I don't have old runs to see if it previously failed or not.  That's why there's a `|| true` at the end of this command